File: DuDoong-Domain/src/main/java/band/gosrock/domain/domains/ticket_item/domain/ItemOptionGroup.java
package band.gosrock.domain.domains.ticket_item.domain;


import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.Table;
import javax.persistence.UniqueConstraint;
import lombok.AccessLevel;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@Table(uniqueConstraints = {@UniqueConstraint(columnNames = {"item_id", "option_group_id"})})
@Entity(name = "tbl_item_option_group")
public class ItemOptionGroup {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "item_option_group_id")
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "item_id", updatable = false)
    private TicketItem item;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "option_group_id", updatable = false)
    private OptionGroup optionGroup;

    @Builder
    public ItemOptionGroup(TicketItem item, OptionGroup optionGroup) {
        this.item = item;
        this.optionGroup = optionGroup;
    }
}

File: DuDoong-Domain/src/main/kotlin/band/gosrock/domain/domains/ticket_item/domain/ItemOptionGroup.kt
package band.gosrock.domain.domains.ticket_item.domain

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.FetchType
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.JoinColumn
import jakarta.persistence.ManyToOne
import jakarta.persistence.Table
import jakarta.persistence.UniqueConstraint

@Table(uniqueConstraints = [UniqueConstraint(columnNames = ["item_id", "option_group_id"])])
@Entity(name = "tbl_item_option_group")
class ItemOptionGroup(
    item: TicketItem? = null,
    optionGroup: OptionGroup? = null,
) {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "item_option_group_id")
    var id: Long? = null
        protected set

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "item_id", updatable = false)
    var item: TicketItem? = item
        protected set

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "option_group_id", updatable = false)
    var optionGroup: OptionGroup? = optionGroup
        protected set
}
